sqlmesh/core/config/connection.py

```diff
@@ -1846,7 +1851,13 @@ def _connection_config_validator(
 ) -> ConnectionConfig | None:
     if v is None or isinstance(v, ConnectionConfig):
         return v
-    return parse_connection_config(v)
+    try:
+        return parse_connection_config(v)
+    except pydantic.ValidationError as e:
+        raise ConfigError(
+            validation_error_message(e, f"Invalid '{v['type']}' connection config:")
+            + "\n\nVerify your config.yaml and environment variables."
```

Comment:
```suggestion
            + "\n\nVerify your config (config.yaml or config.py) and environment variables."
```

Something like this as we have a handful of users use config.py

Reply:
config.py is a different code path. Added error handling for it as well